make/golang.mk

```diff
@@ -27,37 +27,37 @@ _ := $(if  \
 		$(error must be running Go version ^$(1) - you are running $(shell go version | cut -d" " -f3 | cut -c3-)))
 endef
 
-export GO15VENDOREXPERIMENT=1
-
 # FGT is a utility that exits with 1 whenever any stderr/stdout output is recieved.
 FGT := $(GOPATH)/bin/fgt
 $(FGT):
 	go get github.com/GeertJohan/fgt
 
-# Godep is a tool used to manage Golang dependencies in the style of the Go 1.5
-# vendoring experiment.
-GODEP := $(GOPATH)/bin/godep
-$(GODEP):
-	go get -u github.com/tools/godep
+DEP_VERSION = v0.3.2
 
-# Golint is a tool for linting Golang code for common errors.
-GOLINT := $(GOPATH)/bin/golint
-$(GOLINT):
-	go get github.com/golang/lint/golint
+# Dep is a tool used to manage Golang dependencies. It is the offical vendoring experiment, but
+# not yet the official tool for Golang.
+$(GOPATH)/src/github.com/golang/dep:
+	@git clone --branch $(DEP_VERSION) https://github.com/golang/dep.git $(GOPATH)/src/github.com/golang/dep
+$(GOPATH)/bin/dep: $(GOPATH)/src/github.com/golang/dep
+	@git -C $(GOPATH)/src/github.com/golang/dep checkout tags/$(DEP_VERSION)
+	@go build -o $(GOPATH)/bin/dep github.com/golang/dep/cmd/dep
```

Comment:
i'm somewhat partial to the `sfncli` approach: https://github.com/Clever/sfncli/blob/master/make/sfncli.mk

Here it would mean:
- download to a relative `bin/dep` location instead modifying any global path or install of `dep`
- is encoded as a `bin/dep` target
- checks if `bin/dep` is the correct version
- because of this check ^, `bin/dep` is PHONY
- uses github release instead of building from source